src/wallet/wallet_tx_builder.cpp

```diff
@@ -12,153 +13,167 @@ int GetAnchorHeight(const CChain& chain, uint32_t anchorConfirmations)
     return nextBlockHeight - anchorConfirmations;
 }
 
-PrepareTransactionResult WalletTxBuilder::PrepareTransaction(
+ChangeAddress
+WalletTxBuilder::GetChangeAddress(
+        CWallet& wallet,
+        const ZTXOSelector& selector,
+        SpendableInputs& spendable,
+        const Payments& resolvedPayments,
+        const TransactionStrategy& strategy,
+        bool afterNU5) const
+{
+    // Determine the account we're sending from.
+    auto sendFromAccount = wallet.FindAccountForSelector(selector).value_or(ZCASH_LEGACY_ACCOUNT);
+
+    auto getAllowedChangePools = [&](const std::set<ReceiverType>& receiverTypes) {
+        std::set<OutputPool> result{resolvedPayments.GetRecipientPools()};
+        // We always allow shielded change when not sending from the legacy account.
+        if (sendFromAccount != ZCASH_LEGACY_ACCOUNT) {
+            result.insert(OutputPool::Sapling);
```

Comment:
Why only Sapling?

Reply:
This code is only moved in this PR, but I _think_ it was probably new in the initial WalletTxBuilder PR.